lisa/sut_orchestrator/azure/features.py

```diff
@@ -1944,11 +1946,9 @@ def check(self, capability: Any) -> search_space.ResultReason:
             capability, ArchitectureSettings
         ), f"actual: {type(capability)}"
         result = super().check(capability)
-
-        result.merge(
-            search_space.check_setspace(self.arch, capability.arch),
-            "vhd Architecture",
-        )
+        if self.arch != capability.arch:
+            result.result = False
+            result.reasons.append("image arch should be consistent with vm size arch")
```

Comment:
are these changes good?

Reply:
this part is ok, please see my previous comments.